lib/src/Base/Stat/IsotropicCovarianceModel.cxx

```diff
@@ -0,0 +1,208 @@
+//                                               -*- C++ -*-
+/**
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/IsotropicCovarianceModel.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(IsotropicCovarianceModel)
+
+
+/* Parameters constructor */
+IsotropicCovarianceModel::IsotropicCovarianceModel(const CovarianceModel & oneDimensional,
+  const UnsignedInteger inputDimension)
+  : CovarianceModelImplementation(inputDimension)
+{
+  setKernel(oneDimensional);
+  setActiveParameter(kernel_.getActiveParameter());
+  setScale(kernel_.getScale());
+  setAmplitude(kernel_.getAmplitude());
+  setNuggetFactor(kernel_.getNuggetFactor());
```

Comment:
We could move the 4 set calls into setKernel as they depend on the kernel attribute, in case we decide to make setKernel public